src/pages/content-edit/head.js

```diff
@@ -3,261 +3,431 @@ import format from "date-fns/format";
+            ctrl.saving = false;
+            m.redraw();
+        });
+    };
+
+    ctrl.clearSchedule = function() {
+        ctrl.start = ctrl.nulledScheduleStart();
+        ctrl.end   = ctrl.nulledScheduleEnd();
+        ctrl.invalidDates = false;
     };
 
     ctrl.save = function(opts) {
-        ctrl.saving = true;
+        var updated = {};
 
+        ctrl.saving = true;
         m.redraw();
 
-        ref.update({
+        updated = {
             fields : opts.data.fields,
             name   : opts.data.name,
             slug   : opts.data.slug || null
-        }, function() {
-            ctrl.saving = false;
+        };
+
+        updated = ctrl.addScheduleData(updated);
 
+        ref.update(updated, function() {
+            ctrl.saving = false;
             m.redraw();
         });
     };
+
+    ctrl.addScheduleData = function(updated) {
+        if(!ctrl.invalidDates) {
+            let startTs = null,
+                endTs = null,
+                startTime = ctrl.start.time || DEFAULT_START_TIME,
+                endTime = ctrl.end.time || DEFAULT_END_TIME;
+
+            if(ctrl.start.date) {
+                startTs = ctrl.getTimestampFromStr(ctrl.start.date + " " + startTime);
+            }
+            if(ctrl.end.date) {
+                endTs = ctrl.getTimestampFromStr(ctrl.end.date + " " + endTime);
+            }
+
+            updated.published_at = startTs;
+            updated.unpublished_at = endTs;
+
+            if(options.data.published_at && !updated.published_at) {
+                // Publish date was removed. Null it out.
+                updated.published    = null;
+                updated.published_at = null;
+                updated.published_by = null;
+
+                ctrl.start = ctrl.nulledScheduleStart();
+            }
+
+            if(options.data.unpublished_at && !updated.unpublished_at) {
+                // Unpublish date was removed. Null it out.
+                updated.unpublished_at = null;
+                updated.unpublished_by = null;
+
+                ctrl.end = ctrl.nulledScheduleEnd();
+            }
+        }
+
+        return updated;
+    };
+
+
+    ctrl.recalculateTimestamps = function() {
+        var start = ctrl.start,
+            end = ctrl.end;
+
+        if(start.date) {
+            start.ts = ctrl.getTimestampFromStr(start.date + " " + start.time);
+        }
+        if(end.date) {
+            end.ts = ctrl.getTimestampFromStr(end.date + " " + end.time);
+        }
+
+        if(start.date && end.date) {
+            ctrl.invalidDates = (start.ts > end.ts);
+        }
+    };
+
+    ctrl.makeScheduleObj = function(ts) {
+         return {
+            date : format(ts, DATE_FORMAT),
+            time : format(ts, TIME_FORMAT),
+            ts   : ts
+        };
+    };
+
+    ctrl.getTimestampFromStr = function(str) {
+        return parseInt(format(str, "x"), 10);
+    };
+
+    ctrl.nulledScheduleStart = function() {
```

Comment:
I went back and forth on some names, in hindsight I prefer `nulledStartDate`/`nulledEndDate` but I'll just go with `nulledDate` everywhere on your preference.